### PR TITLE
change the file name of the downloaded video to uuid instead of video title

### PR DESCRIPTION
**Pull request by weslayer**

### CODE REVIEW

**Review by NicholasLe04 — APPROVED** [review]
looks good

File: cache.py [hunk 1]
```diff
@@ -48,3 +50,4 @@ def get_video_id(url) -> str:
         parsed_url = urlparse(url)
         video_id = parse_qs(parsed_url.query)['v'][0]
         return(video_id)
+
```
Comment (evanugarte) [inline cache.py:54]:
```suggestion

```

File: cache.py [hunk 2]
```diff
@@ -19,12 +20,13 @@ def add(self, url:str):
...
+        UUID = uuid.uuid4()
+        theuuid = str(UUID)
+        if (theuuid not in os.listdir(self.file_path)):
+            video.download(self.file_path,filename=theuuid+".mp4")
         video_id = self.get_video_id(url)
-        self.video_id_to_path[video_id] = os.path.join(self.file_path, video_file_name)
-        self.current_size_bytes += os.path.getsize(os.path.join(self.file_path, video_file_name))
+        self.video_id_to_path[video_id] = os.path.join(self.file_path, theuuid+".mp4")
+        self.current_size_bytes += os.path.getsize(os.path.join(self.file_path, theuuid +".mp4"))
         self._downsize_cache_to_target_bytes(self.max_size_bytes)
```
Comment (evanugarte) [inline cache.py]: i think what the code should do is 
1. download the file (already in the existing code)
2. rename the downloaded file path to the uuid string (we need to do this)
3. set `self.video_id_to_path[video_id]` to the file name, which we renamed to the uuid


lmk if this makes sense

Comment (weslayer) [reply cache.py]: done

File: cache.py (hunk 1 above)
Comment (weslayer) [reply cache.py:54]: done

File: cache.py [hunk 3]
```diff
@@ -20,9 +21,13 @@ def add(self, url:str):
                         progressive=True,
                     ).order_by("resolution").desc().first()
         video_file_name = video.default_filename
+        UUID = uuid.uuid4()
+        theuuid = str(UUID)
```
Comment (evanugarte) [inline cache.py]:
```suggestion
```

File: cache.py [hunk 4]
```diff
@@ -20,9 +21,13 @@ def add(self, url:str):
...
         if (video_file_name not in os.listdir(self.file_path)):
             video.download(self.file_path)
         video_id = self.get_video_id(url)
+        os.rename(video_file_name, theuuid + ".mp4")
+        video_file_name = theuuid + ".mp4"
```
Comment (evanugarte) [inline cache.py]:
```suggestion
        video_file_name = str(uuid.uuid4()) + ".mp4"
        os.rename(video_file_name, video_file_name)
```

File: cache.py [hunk 5]
```diff
@@ -47,4 +52,4 @@ def clear(self):
     def get_video_id(url) -> str:
         parsed_url = urlparse(url)
         video_id = parse_qs(parsed_url.query)['v'][0]
-        return(video_id)
+        return(video_id)
```
Comment (evanugarte) [inline cache.py]: undo this change, just leave a blank line at the bottom of this file, under the return statement

File: cache.py (hunk 4 above)
Comment (weslayer) [reply cache.py]: image

gives this error when the changes are applied

File: cache.py (hunk 5 above)
Comment (weslayer) [reply cache.py]: done

**Review by evanugarte — APPROVED** [review]
